Sources/PRLabelAnalysis.swift

```diff
@@ -74,4 +74,159 @@ class PRLabelAnalysis {
+            }
+            GithubAPI.editIssue(url: issueData.url, issueEdit: ["title": updatedTitle])
+            // notify of title change
+            GithubAPI.createComment(url: issueData.url,
+                                    comment: "Your title label prefix has been renamed from \(titleLabel) to \(bracketedName)")
+          }
+          break
+        }
+      }
+    }
+    if (labelsToAdd.count > 0) {
+      GithubAPI.addLabelsToIssue(url: issueData.url, labels: Array(Set(labelsToAdd)))
+    } else {
+      GithubAPI.createComment(url: issueData.url,
+                              comment: "The title doesn't have a [Component] prefix.")
+    }
+  }
+
+
+  /// This method receives the incoming pull request data and adds [component] labels based on the
+  /// what files have been modified.
+  ///
+  /// If the [component] title prefix doesn't exist, it writes a comment to notify the submitter, and looks
+  /// at the code diff to figure out if there is a change to only one component and updates the title
+  /// to have that component's label prefix.
+  ///
+  /// If the [component] title prefix isn't entirely accurate, it parses the diff and
+  /// finds if there is a close match and updates the title accordingly. It will also notify the
+  /// submitter of the change.
+  ///
+  /// If there are multiple components being modified, we will comment to let the submitter know
+  /// that multiple components are being modified.
+  ///
+  /// - Parameter PRData: The incoming pull request data
+  class func addAndFixLabelsForPullRequests(PRData: PullRequestData) {
+    var labelsToAdd = [String]()
+    let diffURL = PRData.diff_url
+    let paths = PRLabelAnalysis.getFilePaths(url: diffURL)
+    let labelsFromPaths = Set(PRLabelAnalysis.grabLabelsFromPaths(paths: paths))
+    if (labelsFromPaths.count > 1) {
+      // notify of changing multiple components
+      GithubAPI.createComment(url: PRData.issue_url,
+                              comment: "This PR affects multiple components.")
+    }
+    labelsToAdd.append(contentsOf: labelsFromPaths)
+    if let titleLabel = PRLabelAnalysis.getTitleLabel(title: PRData.title) {
+      // check if there is a title label but it needs fixing.
+      for label in labelsFromPaths {
+        if label == titleLabel {
+          break
+        } else if PRLabelAnalysis.checkIfTwoStringsAreSimilar(str1: titleLabel, str2: label, threshold: 7) {
+          if let range = titleLabel.range(of: "]") {
+            let titleWithoutLabel = titleLabel[range.upperBound...]
+            var updatedTitle = label
+            if titleWithoutLabel.first != " " {
+              updatedTitle += " " + titleWithoutLabel
+            } else {
+              updatedTitle += titleWithoutLabel
+            }
+            if let lastChar = PRData.title.last, lastChar != "." {
+              updatedTitle += "."
+            }
+            GithubAPI.editIssue(url: PRData.issue_url, issueEdit: ["title": updatedTitle])
```

Comment:
Oh - yesterday when I was referring to adding a `.` at the end of the sentence I was referring to the prose that the bot posts back in comments. PR titles don't need to have a `.` at the end (and in fact I don't think we should enforce that).